server/push_notifications.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"fmt"
+	"github.com/mattermost/mattermost/server/public/model"
+)
+
+func (p *Plugin) NotificationWillBePushed(notification *model.PushNotification, userID string) (*model.PushNotification, string) {
+	// We will use our own notifications if:
+	// 1. This is a call start post
+	// 2. We have enabled ringing
+	// 3. The channel is a DM or GM
+	if notification.PostType != "custom_calls" || !*p.getConfiguration().EnableRinging {
```

Comment:
Should we make `custom_calls` a constant since it's used elsewhere already?

Reply:
And one for recording, done.